### ISSUE TITLE
QInput on iOS native requires two taps

### ISSUE THREAD

**Issue by wooliet** [quasarframework/quasar#4337]
### Software version

Operating System - Darwin(18.6.0) - darwin/x64
NodeJs - 8.11.3

Global packages
  NPM - 6.7.0
  yarn - Not installed
  @quasar/cli - 1.0.0-rc.2
  cordova - 9.0.0 (cordova-lib@9.0.1)

Important local packages
quasar - 1.0.0-rc.4
@quasar/app - 1.0.0-rc.6
@quasar/extras - 1.1.2

### What did you get as the error?
No error information. Strictly UX related.

### What were you expecting?
_QInput_ elements on an iOS build seem to require two distinct taps to pull up the keyboard. The first tap gives it focus. The second tap then allows editing the text. 

Expecting a single tap to both bring focus to the input _and_ pull up the keyboard.

### What steps did you take, to get the error?

Any _QInput_ used in a native iOS build. Mobile web is not an issue. Android is not an issue.

Also: not an issue with our 0.17 iOS build.

**Comment by rstoenescu** [comment]
Hi,

Cannot reproduce. Tested both on the emulator and on a real device. Keyboard comes up on first tap.

You haven't specified your Quasar version (only @quasar/app version, but that's the App CLI). Please confirm `quasar` package is 1.0.0-rc.4.

**Comment by webnoob** [comment]
@wooliet I also experienced this issue, however, I can no longer reproduce it.
In addition to the information @rstoenescu has asked for, could you please also post the dependencies section of your src-cordova/package.json file.

I have a feeling that the following sorted it for me:

* npm install -g cordova
* cd src-cordova
* cordova platform remove ios
* cordova platform add ios

I can't be sure and **please** post your package.json data before trying this so we can see if it's an issue with cordova / cordova-ios version or not.

**Comment by wooliet** [comment]
```json
```

**Comment by wooliet** [comment]
Sounds like a good idea @webnoob. I will report back when we've worked through your steps and tested again.

Copied below are the dependencies as they stand now.


_src-cordova/package.json_
```json
  "dependencies": {
    "cordova-android": "^8.0.0",
    "cordova-ios": "^5.0.1",
    "cordova-plugin-camera": "^4.0.3",
    "cordova-plugin-compat": "^1.2.0",
    "cordova-plugin-file": "^6.0.1",
    "cordova-plugin-geolocation": "^4.0.1",
    "cordova-plugin-ios-camera-permissions": "^1.2.0",
    "cordova-plugin-network-information": "^2.0.1",
    "cordova-plugin-splashscreen": "^5.0.2",
    "cordova-plugin-statusbar": "^2.4.2",
    "cordova-plugin-whitelist": "^1.3.3",
    "phonegap-plugin-barcodescanner": "^7.1.2"
  },
```

**Comment by nothingismagick** [comment]
Maybe try with cordova 9

**Comment by wooliet** [comment]
@nothingismagick  I double-checked and that is the version installed:
```bash
npm ls -g --depth=0 | grep cordova
├── cordova@9.0.0
```

**Comment by nothingismagick** [comment]
Also post the results of

quasar info

>

**Comment by nothingismagick** [comment]
Also exactly what devices and what iOS are you using?

**Comment by wooliet** [comment]
vids above on iPhone SE running iOS 12.3.1.  It was reproducible on iPhone X (not sure what versions).

**Comment by webnoob** [comment]
@wooliet Stab in the dark here - Is this an upgraded project from .17 to v1?
If so, does it happen with a brand new project creation using `quasar create <your_proj>` ?

**Comment by wooliet** [comment]
It is an upgrade from 0.17. 

Yes, this happens with a straight out of the box `quasar create`.

_src/pages/Index.vue_
```
<template>
  <q-page padding>
    <q-list no-border>
      <q-item>
        <q-input
          filled
          v-model.trim="nameFirst"
          label="First name"
          stack-label
          class="fit"
        />
      </q-item>
      <q-item>
        <q-input
          filled
          v-model.trim="nameLast"
          label="Last name"
          stack-label
          class="fit"
        />
      </q-item>
    </q-list>
  </q-page>
</template>

<script>
export default {
  name: 'PageIndex',
  data() {
    return {
      nameFirst: null,
      nameLast: null
    };
  }
}
</script>
```

![quasar-input-test](https://user-images.githubusercontent.com/1421908/59625477-00715b80-90ff-11e9-98bb-7ef3a24bad86.gif)

**Comment by nothingismagick** [comment]
Possible related: https://github.com/ionic-team/ionic-v3/issues/1012

**Comment by nothingismagick** [comment]
More likely related: https://github.com/apache/cordova-plugin-wkwebview-engine/issues/86

And a potential fix:
https://www.npmjs.com/package/cordova-plugin-wkwebview-inputfocusfix

**Comment by webnoob** [comment]
Good catch - @wooliet Can you try this and see if it fixes your issue?

**Comment by nothingismagick** [comment]
This might be the issue - we aren't focussing on the same tick...

- you can call .focus() so long as its in the same tick as a user action. For example, you can call .focus() on an element synchronously in click event handler.

- you can call event.preventDefault() on the mousedown event to prevent the input from being blurred

From this PR: https://github.com/apache/cordova-plugin-wkwebview-engine/pull/37/

Also see this:
https://github.com/ionic-team/ionic-plugin-keyboard/issues/274

**Comment by wooliet** [comment]
Any additional guidance? 

Are you (@nothingismagick) saying there are things that can be done within Quasar?

Does the _inputfocus fix_ plugin also rely on installing the _wkwebview engine_ plugin? That seems like a large dependency, and hasn't had a new release in 1.5 years.

**Comment by nothingismagick** [comment]
Try adding this plugin: 
https://www.npmjs.com/package/cordova-plugin-wkwebview-inputfocusfix

And potentially this is something that *could* be fixed, but we need your help to know exactly what the problem is because we can't replicate.

**Comment by dlbass** [comment]
@nothingismagick We had never switched from UIWebView. So, we added WKWebView, and then the inputfocusfix plugin, but the double tap issue persists. 

Not sure if this is related, but we're watching this issue now: https://github.com/onderceylan/cordova-plugin-wkwebview-inputfocusfix/issues/10

We've also tried the ionic webview, but the issue is still there, along with other problems. https://github.com/ionic-team/cordova-plugin-ionic-webview

Our next attempt is to see if we can prove that it is an issue with iOS 12.2 and 12.3.1

**Comment by dlbass** [comment]
We were able to confirm the issue only appears since iOS. Inputs focus as expected in 11.4 :|

**Comment by pdanpdan** [comment]
Can anybody check if this helps?
https://www.outsystems.com/forums/discussion/46409/focus-within-ios-mobile-is-not-opening-keyboard/#Post170328

**Comment by dlbass** [comment]
@pdanpdan We finally had a chance to try the `KeyboardDisplayRequiresUserAction` config mentioned in the Outsystems issue. It did not solve the double-tap issue for us.

**Comment by pdanpdan** [comment]
Thank you for checking.
I think we have a way to solve it.
If you can do it please try to apply this PR (https://github.com/quasarframework/quasar/pull/4659) - it should fix the problem.
But the fix will land in production after Razvan returns from vacation (estimated 3 weeks).

**Comment by dlbass** [comment]
That's excellent news! We'll test it out as soon as we can and report back.

**Comment by JamieMcDonnell** [comment]
Hi guys, any progress on this one? It's holding me back from launching my app. If I can help in any way let me know - I have an iPad and have my app deployed natively onto it. Thanks

**Comment by pdanpdan** [comment]
@JamieMcDonnell @dlbass 

Please make a short test for me - add `"quasar": "pdanpdan/quasar#quasar-v1.0.5-test.9-gitpkg",` in your application's package.json under "dependencies".

Do a yarn && quasar dev and tell me if you can still reproduce the problems.

**Comment by JamieMcDonnell** [comment]
Hay Dan,
Thanks for getting that up - I'll do my best to take a look over the
weekend. Have a good one.
J

**Comment by nicholaszuccarelli** [comment]
Thought I'd have my say here too. I'm having a very similar issue on Cordova iOS, however I don't have to tap twice, but rather long tap on the input.

So if I quickly tap on an input. it will produce the same bug ^, but if I hold down on an input field for half a second, then it will trigger the keyboard / input.

**Comment by JamieMcDonnell** [comment]
Hi Dan,
I got the chance to test this just now. It does seem solved on fields that are not in a qDialog, but for any that are, I see the background page still scrolls as I am trying to focus fields, and I have to tap multiple times in different areas of the field to get focus. Some times focus is given from a double tab, and it seems that sometimes by single tap in a very specific area around the label. Tapping in the area of a textArea for example does not focus the field - only double tapping seems to give it focus.

**Comment by JamieMcDonnell** [comment]
Hi PDan, any progress, questions or assistance I can offer to help get this bug sorted? It is a pretty big show-stopper.

**Comment by pdanpdan** [comment]
I'll do some more tests tomorrow and I'll try to provide another test build.

**Comment by JamieMcDonnell** [comment]
Thanks, let me know if and how I can help get to the bottom of this one.

**Comment by pdanpdan** [comment]
Please make a short test for me - add "quasar": "pdanpdan/quasar#quasar-v1.0.5-test.10-gitpkg", in your application's package.json under "dependencies".

Do a yarn && quasar dev and tell me if you can still reproduce the problems.

**Comment by JamieMcDonnell** [comment]
Hay Dan, I just followed your instructions and have it up and running on the iPad, but I still need to double tap into both q-Input (text and textarea) in order to focus them ;( Is there any way we can collaborate to help you get this resolved?

**Comment by pdanpdan** [comment]
Ok, something is strange, because I tested on iPad and it was ok.
Are you using:
- web
- cordova

If web then what iOS, what browser.

Also I'll need a codepen with the structure you have - if you can do a minimal one that reproduces the problem it would be best.

**Comment by JamieMcDonnell** [comment]
Hay,
I'm using iOS 12.1.4

I see the issue only in the Cordova build

The following fields are in a qDialog
1x qInput (text)
1x qInput (textArea)
2x qSelect.

On the same iPad, Chrome / Safari both are fine, so only Cordova displays the issue.

I will try again to create a codepen that reproduces the bug, but I have done so before and was unable to. Perhaps I can put my project on dropbox and you can test it directly?

**Comment by pdanpdan** [comment]
No, in that case I think it's no need for codepen. But if it's just about Cordova we'll have to wait for @razvan

**Comment by JamieMcDonnell** [comment]
OK, thanks for saving me that bit of time - do please keep me updated so I can keep my client updated ;)

**Comment by ypxio** [comment]
I also have same issue with this. no need to double tap, but single tap (< 0.1s I think) seems like it only doing hover action, then little hold tap trigger focus action to the field.

> Thought I'd have my say here too. I'm having a very similar issue on Cordova iOS, however I don't have to tap twice, but rather long tap on the input.
> 
> So if I quickly tap on an input. it will produce the same bug ^, but if I hold down on an input field for half a second, then it will trigger the keyboard / input.
> 
> Tried on both iPhone and iPad and both produce the same error.

**Comment by rstoenescu** [comment]
Fix will be available in 1.0.6.

**Comment by rstoenescu** [comment]
It was due to a bug in fastclick. We had to fork it. Be sure to update both `@quasar/app` and `quasar` packages (when v1.1.0 is released). Thank you.
